docs/databases/mysql/how-to-create-and-use-mysql-stored-procedures/index.md

```diff
@@ -0,0 +1,322 @@
+---
+author:
+  name: Francis Ndungu
+  email: francisndungu83@gmail.com
+description: 'Learn how to use MySQL stored procedures, which are user-defined SQL statements that are stored in a MySQL database. Learn the syntax for declaring procedures, create your first example procedure, and then learn how to execute it.'
+keywords: ['mysql','stored procedures','database','database optimization']
+license: '[CC BY-ND 4.0](https://creativecommons.org/licenses/by-nd/4.0)'
+published: 2019-02-20
+modified_by:
+  name: Linode
+title: "Creating and Using MySQL Stored Procedures - A Tutorial"
+h1_title: "Creating and Using MySQL Stored Procedures"
+contributor:
+  name: Francis Ndungu
+  link: https://twitter.com/francisndungu83
+external_resources:
+- '[Working with MySQL stored procedures](https://dev.mysql.com/doc/connector-net/en/connector-net-tutorials-stored-procedures.html)'
+---
+
+Stored procedures are user-defined SQL statements that are stored in a MySQL database and executed on-demand to perform a specific database operation. These predefined subroutines help in moving the business logic to the database, which offers a few benefits:
+
+- Round trips made to the database server by an application are minimized.
+- Code maintainability is enhanced, since different applications can share the stored procedure.
+- Database performance is improved, because stored procedures are compiled once and executed very quickly and efficiently.
+
+In this guide, you will:
+
+- Learn the syntax for [creating stored procedures](#creating-a-stored-procedure), and declare your first procedure.
+
+- [Execute the example procedure](#executing-a-stored-procedure) after you've declared it.
+
+- Learn how to [delete a procedure](#deleting-stored-procedures) when you no longer need it.
+
+## Before You Begin
+
+Make sure you have the following:
+
+1.	A configured Linode server. You can learn how to create and setup a Linode server by reading our [Getting Started with Linode](/docs/getting-started/) guide.
+
+1.	A MySQL server and client installed on the Linode server. Installation guides for MySQL are available for different distributions in our [MySQL section](/docs/databases/mysql/).
+
+## Prepare the Database
+
+You will start by creating a sample database, table, and user for accessing the database. You will also populate the table with sample data for testing purposes.
+
+### Creating the Database, Table, and User
+
+1. Log into the MySQL server:
+
+        mysql -u root -p
+
+    You will be prompted to enter the root password of your MySQL database. Then, hit **Enter** to continue.
+
+1.  Next, you will see a MySQL prompt similar to the one shown below.
+
+    {{< output >}}
+mysql >
+{{< /output >}}
+
+1.  Enter the command below to create a `test_db` database:
+
+        Create database test_db;

```

Comment:
We should try to stick to a convention for commands, either we put them in capitals or keep them all lowercase.  A mix are used in this guide.

Reply:
Agreed; changed to all caps across guide, with the exception of user-defined stored procedures